random/README.md

```diff
@@ -0,0 +1,181 @@
+        return WriteError(outErr)
+    case n != 8:
+        outErr := fmt.Errorf("reading random bytes: expected 8 bytes, got %d", n)
+        return WriteError(outErr)
+    }
+
+    roll := (binary.BigEndian.Uint64(randBytes) % input.Sides) + 1
+    return WriteOutput(roll)
+}
+```
+
+You will notice a few things:
+
+- The `rollDie` function is exported so that it can be invoked by the
+  Blocky AS server in a WASM runtime.
+- The function takes two `uint64` arguments and returns a `uint64`. These are
+  fat pointers to shared memory managed by the Blocky AS server, where the first
+  32 bits are a memory address and the second 32 bits are the size of the data.
+  The memory space is sandboxed and shared between the TEE host program (Blocky
+  AS server) and the WASM runtime (your function). The `inputPtr` and
+  `secretPtr` arguments carry user-defined function input and secrets,
+  though we only make use of input in this example. The output of the function
+  is also a memory pointer, whose value will be returned to the user.
+- The function uses the `basm`
+  [Blocky Attestation Service WASM Go SDK](https://github.com/blocky/basm-go-sdk)
+  `basm.ReadFromHost` to fetch the input data and unmarshal it into the
+  `Args` struct.
+- We are able to generate random numbers the way you would in a normal Go
+  program (i.e., via `math/rand` or `crypto/rand`). This is because `tinygo`
+  provides implementations for `math/rand` and `crypto/rand`, and because we
+  properly seed the `wazero` runtime with a source of entropy (`wazero`
+  provides a deterministic source of entropy by default).
+
+### Step 2: Compile the function to WebAssembly (WASM)
+
+To invoke our function in the Blocky AS server, we first need to compile
+it into a WASM file. If you inspect the `build` target in the
+[`Makefile`](./Makefile), you'll see the build command:
+
+```bash
+@docker run --rm \
+    -v .:/src \
+    -w /src \
+    tinygo/tinygo:0.31.2 \
+    tinygo build -o tmp/x.wasm -target=wasi ./...
+```
+
+where we use `docker` to run [`tinygo`](https://tinygo.org/) to compile
+[`main.go`](./main.go) to WASM and save it to `tmp/x.wasm`. You can build our
+function by calling:
+
+```bash
+make build
+```
+
+### Step 3: Invoke the function on the Blocky AS server
+
+To invoke the `rollDie` example, call:
+
+```bash
+make run
+```
+
+You will see the following output extracted from a Blocky AS response:
+
+```
+Output:
+{
+  "Success": true,
+  "Error": "",
+  "Value": 3
+}
+```
+
+You will most likely see a different number for the `Value` field as it is
+randomly generated. On that note, subsequent invocations of `rollDie` should
+produce different numbers. You can control the range of said numbers by
+updating the `sides` field in the `fn-call.json` file.
+
+## Next steps
+
+Now that you have successfully run the example, you can start modifying it to
+fit your own needs. Check out other examples in this repository, to learn what
```

Comment:
One thing you could ask people to do is extend the example with a message that says "Critical Failure" or "Critical Success". You could also suggest that they roll multiple different sided dies in the same call, since that's often what's needed in D&G (though I think D&G maybe trademarked, so we may want to be somewhat generic about that suggestion).